src/cdk/table/array-data-source.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {DataSource} from './data-source';
+import {BehaviorSubject} from 'rxjs/BehaviorSubject';
+
+/**
+ * Array-based data source that can be used to connect to a CollectionViewer. Will trigger an
+ * update to the CollectionViewer whenever a new data array is set.
+ */
+export class ArrayDataSource<T> implements DataSource<T> {
+  /** Stream that emits when a new data array is set on the data source. */
+  private _data: BehaviorSubject<T[]>;
+  set data(data: T[]) { this._data.next(data); }
+  get data() { return this._data.value; }
+
+  constructor(initialData: T[] = []) {
+    this._data = new BehaviorSubject<T[]>(initialData);
+  }
+
+  /**
+   * Triggers an update to the CollectionViewer. Should be used when elements are added, removed,
+   * or moved in the data array and should be reflected in the CollectionViewer.
+   */
+  update() { this._data.next(this.data); }
```

Comment:
This `update` function would only be necessary for users who want to set their data once and expect the table to always be based off the same array reference. For those who would say `arrayDataSource.data = newArraySlice` to update their data, they won't need this function.

I'm anticipating that this will be an easier concept for those who want to manipulate a single data array and see those changes propagated to the table. They'll wonder why they need to keep setting `data` to the same array just to see an update.

I'm open to your suggestions on whether or not this would be useful. Do you think people would be alright if we told them to do `arrayDataSource.data = theirData` each time they want to render an update?